src/primitives/felt.ts

```diff
@@ -1,61 +1,54 @@
 import { MaybeRelocatable, Relocatable } from './relocatable';
 
-import {
-  ForbiddenOperation,
-  OutOfRangeBigInt,
-  PrimitiveError,
-} from 'errors/primitives';
+import { ForbiddenOperation, OutOfRangeBigInt } from 'errors/primitives';
 
 export class Felt {
   private inner: bigint;
   static PRIME: bigint =
     0x800000000000011000000000000000000000000000000000000000000000001n;
   static ZERO: Felt = new Felt(0n);
   constructor(_inner: bigint) {
-    if (_inner < 0n || _inner > Felt.PRIME) {
-      throw new PrimitiveError(OutOfRangeBigInt);
-    }
-    this.inner = _inner;
+    while (_inner < 0n) _inner += Felt.PRIME;
+    this.inner = _inner % Felt.PRIME;
   }
 
   add(other: MaybeRelocatable): Felt {
     if (!(other instanceof Felt)) {
-      throw new PrimitiveError(ForbiddenOperation);
+      throw new ForbiddenOperation();
     }
-    return new Felt((this.inner + other.inner) % Felt.PRIME);
+
+    return new Felt(this.inner + other.inner);
   }
 
   sub(other: MaybeRelocatable): Felt {
     if (!(other instanceof Felt)) {
-      throw new PrimitiveError(ForbiddenOperation);
+      throw new ForbiddenOperation();
     }
 
-    let result = this.inner - other.inner;
-    if (result < 0n) {
-      result += Felt.PRIME;
-    }
-    return new Felt(result);
+    return new Felt(this.inner - other.inner);
   }
 
   mul(other: MaybeRelocatable): Felt {
     if (!(other instanceof Felt)) {
-      throw new PrimitiveError(ForbiddenOperation);
+      throw new ForbiddenOperation();
     }
-    return new Felt((this.inner * other.inner) % Felt.PRIME);
+
+    return new Felt(this.inner * other.inner);
   }
 
   div(other: MaybeRelocatable): Felt {
     if (!(other instanceof Felt) || other.inner === 0n) {
-      throw new PrimitiveError(ForbiddenOperation);
+      throw new ForbiddenOperation();
     }
-    let result = this.inner / other.inner;
-    return new Felt(result);
+
+    return new Felt(this.inner / other.inner);
```

Comment:
this is not the proper felt division, which should return x such that `other * x = inner`

Reply:
Indeed, it is not a proper felt division, which should be the multiplicative inverse:
$a / b = a * b^{-1} = x$

```suggestion
const inv_other = other.inv()
return this.inner.mul(inv_other)
```

With `inv()` being a method to compute the inverse of a felt. The easiest & widely known algorithm to implement is the extended euclidean algorithm